### PR TITLE
[AMS] Move aws sdk dependency profile

### PR DESCRIPTION
**Pull request by XBaith**
## Why are the changes needed?

Fix #1479

## How was this patch tested?
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

  - Does this pull request introduces a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Review by baiyangtx — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -594,6 +600,49 @@
                 <hive.version>2.1.1</hive.version>
             </properties>
         </profile>
+        <profile>
+            <id>aws</id>
+            <activation>
+                <activeByDefault>false</activeByDefault>
+            </activation>
+            <dependencies>
```
Comment (baiyangtx) [inline pom.xml:608]: This should be dependencyManagement.

Comment (shidayang) [inline pom.xml:608]: This will cause all module has these dependencies

**Review by baiyangtx — CHANGES_REQUESTED** [review]
The dependencies shoud be added to core/pom.xml and manager the version in /pom.xml via dependencyManagement.

**Review by shidayang — CHANGES_REQUESTED** [review]

File: core/pom.xml [hunk 2]
```diff
@@ -88,6 +88,11 @@
             <artifactId>iceberg-parquet</artifactId>
         </dependency>
 
+        <dependency>
```
Comment (shidayang) [inline core/pom.xml:91]: Remove this dependency

**Review by shidayang — CHANGES_REQUESTED** [review]

File: core/pom.xml (hunk 2 above)
Comment (XBaith) [reply core/pom.xml]: Don

**Review by baiyangtx — APPROVED** [review]

**Pull request merged**